parsely/src/test/java/com/parsely/parselyandroid/EventsBuilderTest.kt

```diff
@@ -0,0 +1,216 @@
+            "heartbeat",
+            null,
+            null,
+            TEST_UUID,
+        )
+
+        // then
+        assertThat(event)
+            .containsEntry("action", "heartbeat")
+            .containsEntry("pvid", TEST_UUID)
+            .sharedPixelAssertions()
+    }
+
+    @Test
+    fun `events builder prepares correct videostart pixel`() {
+        // when
+        val event: Map<String, Any> = sut.buildEvent(
+            TEST_URL,
+            "",
+            "videostart",
+            null,
+            null,
+            TEST_UUID,
+        )
+
+        // then
+        assertThat(event)
+            .containsEntry("action", "videostart")
+            .containsEntry("vsid", TEST_UUID)
+            .sharedPixelAssertions()
+    }
+
+    @Test
+    fun `events builder prepares correct vheartbeat pixel`() {
+        // when
+        val event: Map<String, Any> = sut.buildEvent(
+            TEST_URL,
+            "",
+            "vheartbeat",
+            null,
+            null,
+            TEST_UUID,
+        )
+
+        // then
+        assertThat(event)
+            .containsEntry("action", "vheartbeat")
+            .containsEntry("vsid", TEST_UUID)
+            .sharedPixelAssertions()
+    }
+
+    @Test
+    fun `given extraData is null, when creating a pixel, don't include extraData`() {
+        // given
+        val extraData: Map<String, Any>? = null
+
+        // when
+        val event: Map<String, Any> = sut.buildEvent(
+            TEST_URL,
+            "",
+            "pageview",
+            null,
+            extraData,
+            TEST_UUID,
+        )
+
+        // then
+        @Suppress("UNCHECKED_CAST")
+        assertThat(event["data"] as Map<String, Any>).hasSize(5)
+    }
+
+    @Test
+    fun `given extraData is not null, when creating a pixel, include extraData`() {
+        // given
+        val extraData: Map<String, Any> = mapOf(
+            "extra 1" to "data 1",
+            "extra 2" to "data 2"
+        )
+
+        // when
+        val event: Map<String, Any> = sut.buildEvent(
+            TEST_URL,
+            "",
+            "pageview",
+            null,
+            extraData,
+            TEST_UUID,
+        )
+
+        // then
+        @Suppress("UNCHECKED_CAST")
+        assertThat(event["data"] as Map<String, Any>).hasSize(7)
+            .containsAllEntriesOf(extraData)
+    }
+
+    @Test
+    fun `given metadata is null, when creating a pixel, don't include metadata`() {
+        // given
+        val metadata: ParselyMetadata? = null
+
+        // when
+        val event: Map<String, Any> = sut.buildEvent(
+            TEST_URL,
+            "",
+            "pageview",
+            metadata,
+            null,
+            TEST_UUID,
+        )
+
+        // then
+        assertThat(event).doesNotContainKey("metadata")
+    }
+
+    @Test
+    fun `given metadata is not null, when creating a pixel, include metadata`() {
+        // given
+        val metadata =
```

Comment:
Minor (🔍): Just as an FYI, I would usually I split this like this, just to keep the connection between the parameter name and the type on the same line and have all the parameters on a separate line when/if needed:

```kotlin
        val metadata = ParselyMetadata(
            ArrayList<String>(), "link", "section", null, null, null, null
        )
```

PS: Very very very ultra duper very minor... 😛